Bump xi_unicode to latest version

jschwe opened the pull request:
There is no changelog available, but we only use `linebreak_property` and `LineBreakLeafiter` from `xi_unicode`, so it should be safe to update,

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `___` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

jschwe committed:
Bump xi_unicode to latest version

Signed-off-by: Jonathan Schwender <jonathan.schwender@huawei.com>

mukilan reviewed (approved):
Thanks!

mukilan commented:
It seems like the two test failures are due to breaking change in xi-unicode. The linebreak_property function now returns 42 instead of 40 for the codepoint 'u200D'

https://github.com/xi-editor/xi-editor/commit/0c82f2eb8c508204cf2e7560047b8e060361eaed#diff-c1bc7323d4980a7cfbd7700f185ea3a855e998af62e7cdd43e5d9985080e96e8R471

mukilan commented:
It looks like just a bug in xi-unicode where the numeric value of the XI_LINE_BREAKING_CLASS_ZWJ constant has not been updated to be 42 instead of 40. The `[U_LB_ZWJ](https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/uchar_8h.html#a5d1abdf05be22cb9599f804a8506277cacf051a491922f822f16cfc66efda7f3d)` constant also has a [numeric value of 42](https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/uchar_8h_source.html#l02428)

mukilan committed:
dep: update XI_LINE_BREAKING_CLASS_ZWJ constant to 42

This matches the value returned by xi-unicode's linebreak_property
for `u200D` codepoint in v0.3.0 of the crate.

Signed-off-by: Mukilan Thiyagarajan <mukilan@igalia.com>

mrobinson commented:
> It looks like just ~a bug in xi-unicode~ where the numeric value of the XI_LINE_BREAKING_CLASS_ZWJ constant has not been updated to be 42 instead of 40. The [U_LB_ZWJ](https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/uchar_8h.html#a5d1abdf05be22cb9599f804a8506277cacf051a491922f822f16cfc66efda7f3d) constant also has a [numeric value of 42](https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/uchar_8h_source.html#l02428)
> 
> edit: XI_LINE_BREAKING_CLASS_ZWJ is defined in servo, not xi-unicode so we can update it in this PR.

Nice investigation. We should really stop using these undocumented xi-unicode classes.
